cpp/tests/utilities_tests/lists_column_wrapper_tests.cu

```diff
@@ -0,0 +1,722 @@
+TYPED_TEST_CASE(ListColumnWrapperTestTyped, FixedWidthTypesNoBool);
+
+TEST_F(ListColumnWrapperTest, ListOfInts)
+{
+  using namespace cudf;
+
+  // List<int>, 1 row
+  //
+  // List<int32_t>:
+  // Length : 1
+  // Offsets : 0, 2
+  // Children :
+  //    2, 3
+  //
+  {
+    test::lists_column_wrapper list{2, 3};
+
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 1);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 2);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 2);
+    test::fixed_width_column_wrapper<int> e_data({2, 3});
+    test::expect_columns_equal(e_data, data);
+  }
+
+  // List<int>, 1 row
+  //
+  // List<int32_t>:
+  // Length : 1
+  // Offsets : 0, 2
+  // Children :
+  //    2, 3
+  //
+  {
+    test::lists_column_wrapper list{{2, 3}};
+
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 1);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 2);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 2);
+    test::fixed_width_column_wrapper<int> e_data({2, 3});
+    test::expect_columns_equal(e_data, data);
+  }
+}
+
+TEST_F(ListColumnWrapperTest, ListOfIntsWithValidity)
+{
+  using namespace cudf;
+
+  auto valids = cudf::test::make_counting_transform_iterator(
+    0, [](auto i) { return i % 2 == 0 ? true : false; });
+
+  // List<int>, 1 row
+  //
+  // List<int32_t>:
+  // Length : 1
+  // Offsets : 0, 2
+  // Children :
+  //    2, NULL
+  //
+  {
+    test::lists_column_wrapper list{{{2, 3}, valids}};
+
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 1);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 2);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 2);
+    test::fixed_width_column_wrapper<int> e_data({2, 3}, valids);
+    test::expect_columns_equal(e_data, data);
+  }
+
+  // List<int>, 3 rows
+  //
+  // List<int32_t>:
+  // Length : 3
+  // Offsets : 0, 2, 4, 7
+  // Children :
+  //    2, NULL, 4, NULL, 6, NULL, 8
+  {
+    test::lists_column_wrapper list{{{2, 3}, valids}, {{4, 5}, valids}, {{6, 7, 8}, valids}};
+
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 3);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 4);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2, 4, 7});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 7);
+    test::fixed_width_column_wrapper<int> e_data({2, 3, 4, 5, 6, 7, 8}, valids);
+    test::expect_columns_equal(e_data, data);
+  }
+}
+
```

Comment:
It would be good to add tests for empty lists and also a list with all null entries.
We've had issues in the past printing compound columns like these.
I'm not sure the best way to verify these. Perhaps just call to_string and check the returned string.